catlearn/graph_utils.py

```diff
@@ -750,6 +752,10 @@ def random_walk_edge_sample(
     * use_opposite=True, use_both_ends=True:
       -> candidates are 2->3, 5->2, 1->4 and 0->1
     """
+    def get_degree_safe(graph, node, in_degree):
+        degrees = graph.in_degree([node]) if in_degree else graph.out_degree([node])
+        return dict(degrees).get(node, 0)  # thank you networkx...
```

Comment:
The comment is somewhat sarcastic, but only as a reference to networkkx quirky API on this usage. If I pass a single vertex, I have either a int if the node exstis, or a class otherwise. And then the resulting `DegreeView` has dict-like access but no method to return a default value. I can add extra comments for clarification tough.

I could have used exceptions, but I am personally biased against them, even though they are native python constructs x)

Reply:
Updated